### ISSUE TITLE
Compatibility with Hashie 3.5.0

### ISSUE THREAD

**Issue by bodrovis** [zquestz/omniauth-google-oauth2#265]
Hashie 3.5.0 is released today and seems like this PR https://github.com/intridea/hashie/pull/377 causes omniauth-google-oauth2 to stop working. Here is an error:

```
F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:94:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'omniauth-google-oauth2'. (Bundler::GemRequireError)
Gem Load Error is: uninitialized constant Hashie::Extensions::RubyVersionCheck::ClassMethods::RubyVersion
Backtrace for gem load error is:
F:/Ruby225/lib/ruby/gems/2.2.0/gems/hashie-3.5.0/lib/hashie/extensions/ruby_version_check.rb:10:in `with_minimum_ruby'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/hashie-3.5.0/lib/hashie/array.rb:8:in `<class:Array>' F:/Ruby225/lib/ruby/gems/2.2.0/gems/hashie-3.5.0/lib/hashie/array.rb:5:in `<module:Hashie>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/hashie-3.5.0/lib/hashie/array.rb:4:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/hashie-3.5.0/lib/hashie/mash.rb:2:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-1.3.2/lib/omniauth/strategy.rb:1:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-oauth2-1.4.0/lib/omniauth/strategies/oauth2.rb:15:in `<class:OAuth2>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-oauth2-1.4.0/lib/omniauth/strategies/oauth2.rb:14:in `<module:Strategies>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-oauth2-1.4.0/lib/omniauth/strategies/oauth2.rb:8:in `<module:OmniAuth>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-oauth2-1.4.0/lib/omniauth/strategies/oauth2.rb:7:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-google-oauth2-0.4.1/lib/omniauth/strategies/google_oauth2.rb:3:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-google-oauth2-0.4.1/lib/omniauth/google_oauth2.rb:1:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/omniauth-google-oauth2-0.4.1/lib/omniauth-google-oauth2.rb:1:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:91:in `require' F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:91:in `block (2 levels) in require'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:86:in `each'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:86:in `block in require'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:75:in `each'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:75:in `require' F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler.rb:107:in `require'
f:/rails/my/freelance-de/config/application.rb:16:in `<top (required)>'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:88:in `require'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:88:in `block in server'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:85:in `tap'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:85:in `server'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands.rb:18:in `<top (required)>'
bin/rails:4:in `require'
bin/rails:4:in `<main>'
Bundler Error Backtrace:
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:90:in `block (2 levels) in require'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:86:in `each'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:86:in `block in require'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:75:in `each'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler/runtime.rb:75:in `require'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/bundler-1.14.3/lib/bundler.rb:107:in `require'
        from f:/rails/my/freelance-de/config/application.rb:16:in `<top (required)>'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:88:in `require'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:88:in `block in server'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:85:in `tap'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:85:in `server'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
        from F:/Ruby225/lib/ruby/gems/2.2.0/gems/railties-5.0.1/lib/rails/commands.rb:18:in `<top (required)>'
        from bin/rails:4:in `require'
        from bin/rails:4:in `<main>'
```

Reverting to Hashie 3.4.6 fixes this problem.

**Comment by jonhue** [comment]
It's true. It's not great.

**Comment by bmclean** [comment]
Hashie [3.5.1 ](https://github.com/intridea/hashie/pull/392) been released with a fix.

**Comment by bodrovis** [comment]
This seems to fix the problem, though now I'm getting a warning. This warning, however, seems to be related to Hashie, so I'll continue discussion here https://github.com/intridea/hashie/issues/391#issuecomment-276490533

**Comment by michaelherold** [comment]
I just released Hashie v3.5.2 which has the `disable_warnings` feature for the Mash logger and sets the Hashie logger to the Rails logger when we detect that we're running in Rails. Hopefully, this helps!

**Comment by jonhue** [comment]
@michaelherold Hashie 3.5.2 seems to have some big Issues which prevent the rails server from starting.
intridea/hashie#407
